src/drivers/intel/ssp/mn.c

```diff
@@ -173,6 +185,20 @@ static inline int check_current_mclk_source(uint32_t mclk_rate)
 		ret = -EINVAL;
 	}
 
+	/* if the mclk is already used, can't change its divider, just increase ref count */
+	if (mn->mclk_sources_ref[mclk_id] > 0) {
+		tr_warn(&mn_tr, "MCLK %d is used for multiple SSPs", mclk_id);
```

Comment:
ok, but please also tell the user what the outcome may be  e.g. audio artifacts, or poor audio performance etc.  This way they can map this error to what they hear.

Reply:
maybe I should change it to info level, this doesn't mean audio artifacts necessary, what we need to make sure is that those SSPs are asking for exactly same MCLK output.

Reply:
info is fine, as long as a novice user/dev can read the log and understand this is why performance is poor.

Reply:
> maybe I should change it to info level, this doesn't mean audio artifacts necessary, what we need to make sure is that those SSPs are asking for exactly same MCLK output.

As you wrote Keyon, we should check if the SSP is asking for exactly the same MCLK output. If yes then print info message or even none, but if it is different then we definitely add an error here as this indicate invalid configuration.

Reply:
@mwasko good point, will change.